cmd/common_test.go

```diff
@@ -1,71 +1,201 @@
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 )
 
-// Hook up gocheck into the "go test" runner.
-func Test(t *testing.T) {
-	check.TestingT(t)
-}
+func TestParseClusterAndProjectID(t *testing.T) {
+	t.Parallel()
 
-type CommonTestSuite struct {
-}
+	tests := []struct {
+		id,
+		cluster string
+		project   string
+		shouldErr bool
+	}{
+		{
+			id:      "local:p-12345",
+			cluster: "local",
+			project: "p-12345",
+		},
+		{
+			id:      "c-12345:p-12345",
+			cluster: "c-12345",
+			project: "p-12345",
+		},
+		{
+			id:        "cocal:p-12345",
+			shouldErr: true,
+		},
+		{
+			id:        "c-123:p-123",
+			shouldErr: true,
+		},
+		{
+			shouldErr: true,
+		},
+		{
+			id:      "c-m-12345678:p-12345",
+			cluster: "c-m-12345678",
+			project: "p-12345",
+		},
+		{
+			id:        "c-m-123:p-12345",
+			shouldErr: true,
+		},
+	}
 
-var _ = check.Suite(&CommonTestSuite{})
+	for _, test := range tests {
+		test := test
+		t.Run(test.id, func(t *testing.T) {
+			t.Parallel()
 
-func (s *CommonTestSuite) SetUpSuite(c *check.C) {
+			cluster, project, err := parseClusterAndProjectID(test.id)
+			if test.shouldErr {
+				require.Error(t, err)
+				return
+			}
 
+			require.NoError(t, err)
+			assert.Equal(t, test.cluster, cluster)
+			assert.Equal(t, test.project, project)
+		})
+	}
 }
 
-func (s *CommonTestSuite) TestParseClusterAndProjectID(c *check.C) {
-	testParse(c, "local:p-12345", "local", "p-12345", false)
-	testParse(c, "c-12345:p-12345", "c-12345", "p-12345", false)
-	testParse(c, "cocal:p-12345", "", "", true)
-	testParse(c, "c-123:p-123", "", "", true)
-	testParse(c, "", "", "", true)
-	testParse(c, "c-m-12345678:p-12345", "c-m-12345678", "p-12345", false)
-	testParse(c, "c-m-123:p-12345", "", "", true)
-}
+func TestConvertSnakeCaseKeysToCamelCase(t *testing.T) {
+	t.Parallel()
 
-func (s *CommonTestSuite) TestConvertSnakeCaseKeysToCamelCase(c *check.C) {
-	cases := []struct {
-		input   map[string]interface{}
-		renamed map[string]interface{}
+	tests := []struct {
+		input map[string]any
+		want  map[string]any
 	}{
 		{
-			map[string]interface{}{"foo_bar": "hello"},
-			map[string]interface{}{"fooBar": "hello"},
+			map[string]any{"foo_bar": "hello"},
+			map[string]any{"fooBar": "hello"},
+		},
+		{
+			map[string]any{"fooBar": "hello"},
+			map[string]any{"fooBar": "hello"},
+		},
+		{
+			map[string]any{"foobar": "hello", "some_key": "valueUnmodified", "bar-baz": "bar-baz"},
+			map[string]any{"foobar": "hello", "someKey": "valueUnmodified", "bar-baz": "bar-baz"},
 		},
 		{
-			map[string]interface{}{"fooBar": "hello"},
-			map[string]interface{}{"fooBar": "hello"},
+			map[string]any{"foo_bar": "hello", "backup_config": map[string]any{"hello_world": true}, "config_id": 123},
+			map[string]any{"fooBar": "hello", "backupConfig": map[string]any{"helloWorld": true}, "configId": 123},
 		},
+	}
+
+	for _, test := range tests {
+		test := test
+		t.Run("", func(t *testing.T) {
```

Comment:
Maybe instead of "" here, we could output the test number?

```go
for i, test := range tests {
    test := test
    t.Run(fmt.Sprintf("TestConvertSnakeCaseKeysToCamelCase-%d", i), func(...) { 
    })
}
```

Reply:
Yep, just the number will do. There is no need to pass the test name. Go subtests always carry the parent test's name in the output.